### DISCUSSION TITLE
Module usage metrics?

### DISCUSSION THREAD

**Discussion by vicary** [denoland/deno#9644] [General]
So basically it is the weekly downloads in npm that I am thinking of.

When there are multiple alternatives, the number of active dependents is an important indicator when choosing a library during the design phase of every project.

1. https://deno.land/x by design should not have this info
2. GitHub Traffics tab is only accessible to repo authors/maintainers, and it only shows repo level stats, instead of the single file import that deno wants.

I am curious on how developers, especially deno developers, should be assessing third party modules?

**Comment by wperron** [comment]
There's an issue on the registry repo that you can subscribe to that addresses the download counts https://github.com/denoland/deno_registry2/issues/57. Getting a view of the dependents is a tougher one, it might get done on deno.land/x at some point, but probably not in the near future.

As for assessing whether a third party module is good for you, I suggest going through the github repo and looking through it a bit; are there regular commits? how many contributors does it have? how many releases? how frequent are the releases? These questions will probably get you a better sense of the quality of a third party module than download counts or dependents list could.

**Reply by vicary** [reply]
Thanks for sharing the issue, I'm following it.

Yes, development activity is one of the many facts in my checklist, but that's only part of the story.

In fact it highly depends on which stage of a lifecycle it is. i.e. It should be active at mid-early development stage, but it may not be chosen for a project because of stability requirements; while, when feature complete, commits must be scarce with a lot of active dependents.

Rather than specific feature planned or in progress, I just want to know how Deno treat this area in a broad sense.

**Reply by kitsonk** [reply]
Deno is an open platform and doesn't want to sit being judge and jury on what software people run on that platform. It is for the wider community to help foster what is "good" and "bad". Node.js and npm are tightly coupled together and some would argue that has not been good for innovation and evolution of the ecosystem.

**Reply by vicary** [reply]
@kitsonk Do you mean Deno encourages third party registries with specific features that they see fit for the community, or by community you mean we all discuss here and add features in https://deno.land/x?
